ingeniamotion/wizard_tests/stoppable.py

```diff
@@ -24,16 +24,21 @@ def wrapper(self, *args, **kwargs):  # type: ignore
         return wrapper
 
     def reset_stop(self) -> None:
-        self.is_stopped = False
+        if self.stop_queue.full():
+            self.stop_queue.get(block=False)
 
     def stop(self) -> None:
-        self.is_stopped = True
+        if not self.stop_queue.full():
+            self.stop_queue.put(StopException(), block=False)
 
     def check_stop(self) -> None:
-        if self.is_stopped:
-            raise StopException
+        if self.stop_queue.full():
+            raise self.stop_queue.get(block=False)
```

Comment:
First checking the list and then getting the item could create a race condition where get(block=False) would return None and raise None.
Simple solution:

```
def check_stop(self) -> None:
  ex = self.stop_queue.get(block=False)
  
  if ex is not None:
      raise ex
```
Also, mypy is not complaining that you could raise None? isn't strictness set to True for this project??

Reply:
Mypy is not complaining because the get method raises an Empty exception if the queue is empy instead of returning None. That's why I'm checking if the queue is full before calling the get method. But I see your point, I'll fix it.

https://docs.python.org/3/library/queue.html#queue.Queue.get

Reply:
@polfeliu Done [fe1299d](https://github.com/ingeniamc/ingeniamotion/pull/177/commits/fe1299dffcffa1751c4b9862a4b49d88d2512d25)